### PR TITLE
feat: build e2e tests for binary sync

### PR DESCRIPTION
**Pull request by marekfexa-outreach**
## What this PR does / why we need it
- Builds as part of `make devspace` e2e test packages
- These packages are then available in devspace pod in binary sync mode without any build infrastructure for test execution 

## Jira ID

[QF-880]

## Notes for your reviewers

[QF-880]: https://outreach-io.atlassian.net/browse/QF-880?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: root/Magefile.go [hunk 1]
```diff
@@ -30,6 +31,46 @@ func Version() {
...
+	}
+
+	buildDir, err := ensureBinExists(cwd)
+	if err != nil {
+		return err
+	}
+
+	e2ePackages, err := GetE2eTestPaths(".")
+	if err != nil {
+		return errors.Wrap(err, "Unable to find e2e pacakges")
```
Comment (tomassusovsky-outreach) [inline root/Magefile.go:48]:
```suggestion
		return errors.Wrap(err, "Unable to find e2e packages")
```

File: root/dir.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+			if filepath.Ext(file.Name()) == ".go" {
+				filePath := filepath.Join(path, file.Name())
+				contentBytes, err := ioutil.ReadFile(filePath)
+				if err != nil {
+					return err
+				}
+				content := string(contentBytes)
+
+				// We care fore packages that has at least one test entrypoint in file with or_e2e tag
```
Comment (tomassusovsky-outreach) [inline root/dir.go:47]:
```suggestion
				// We care for packages that has at least one test entrypoint in file with or_e2e tag
```

File: e2e/e2e.go [hunk 3]
```diff
@@ -215,6 +219,40 @@ func shouldRunE2ETests() (bool, error) {
...
+	if err := osStdInOutErr(exec.CommandContext(ctx, "make", "devspace")).Run(); err != nil {
+		return errors.Wrap(err, "Failed to build binaries for devspace")
+	}
+
+	log.Info().Msg("Starting devspace pod and running e2e tests")
+	if err := osStdInOutErr(exec.CommandContext(ctx, "devenv", "--skip-update", "apps", "e2e", "--sync-binaries", ".")).Run(); err != nil {
+		return errors.Wrapf(err, "Failed to deploy %s into devenv", serviceName)
+	}
+	return nil
+	//TODO(marekfexa-outreach): status code handling
```
Comment (marekfexa-outreach) [inline e2e/e2e.go:253]: This code is behind FF USE_DEVSPACE so I think we have it in main like this for few days.

File: e2e/e2e.go [hunk 4]
```diff
@@ -27,6 +27,10 @@ import (
 // flagship is the name of the flagship
 const flagship = "flagship"
 
+// devenvAlreadyExists contains message when devenv exists
+const devenvAlreadyExists = "Re-using existing cluster, this may lead to a non-reproducible failure/success. " +
+	"To ensure a clean operation, run `devenv destroy` before running tests)"
```
Comment (tomassusovsky-outreach) [inline e2e/e2e.go:32]:
```suggestion
	"To ensure a clean operation, run `devenv destroy` before running tests"
```

**Review by tomassusovsky-outreach — APPROVED** [review]

**Pull request merged**